addons/playersuppression/functions/fnc_main.sqf

```diff
@@ -0,0 +1,25 @@
+/*
+ * Author: LAxemann
+ * Checks how much the suppression value should decrease, should be
+ * called via the CBA perFrameHandler with 1s delay
+ *
+ * Example:
+ * [ace_playersuppression_fnc_main, 1, []] call CBA_fnc_addPerFrameHandler
+ *
+ * Public: Yes
+ */
+#include "script_component.hpp"
+
+if (GVAR(threshold) > 0) then {
+    private ["_shotAt","_subtract"];
+    _shotAt 	= false;	
+    if ((time - GVAR(lastShotAt)) <= 1.75) then {_shotAt = true};

```

Comment:
In the future, these GVAR's could be variables on the unit instead of globally; it would allow for suppression to be utilized on AI as well. If its not trouble, we should change these to getVariable's on the unit.